### PR TITLE
Authors metadata

### PR DESCRIPTION
**Pull request by jcubit**
Added metadata field to the manifest

### CODE REVIEW

File: Tests/SPIManifestTests/ManifestTests.swift [hunk 1]
```diff
@@ -25,12 +25,14 @@ class ManifestTests: XCTestCase {
     }
 
     func test_encode_manifest() throws {
-        let m = Manifest(builder: .init(configs: [
+        let m = Manifest(metadata: .init(authors: "Christian Noon, Jon Shier and other contributors"), builder: .init(configs: [
```
Comment (daveverwer) [inline Tests/SPIManifestTests/ManifestTests.swift:28]: Let's make author test data anonymous throughout all tests. "Author One and Author Two" should be good unless testing the max length of the string. Then some repetitions of "0123456789". 👍

File: Tests/SPIManifestTests/ManifestTests.swift [hunk 2]
```diff
@@ -68,19 +102,21 @@ class ManifestTests: XCTestCase {
     func test_load_maxSize() throws {
         let data = Data("""
                 version: 1
+                metadata:
+                  authors: Christian Noon, Jon Shier and other contributors
```
Comment (daveverwer) [inline Tests/SPIManifestTests/ManifestTests.swift:106]: I don't think we need to modify this test as we add this key. The test checks that for a predefined set of data that the `maxByteSize` parameter of `load` succeeds/fails. We should look at how we call this in production to make sure that we have enough capacity for this extra data, but I don't think this test needs to change.

File: Tests/SPIManifestTests/ManifestTests.swift [hunk 3]
```diff
@@ -58,7 +62,37 @@ class ManifestTests: XCTestCase {
...
-                       Manifest(builder: .init(configs: [
+                       Manifest(metadata: .init(authors: "Christian Noon, Jon Shier and other contributors"),
+                                builder: .init(configs: [
+                        .init(platform: Platform.watchos.rawValue,
+                              scheme: "Alamofire watchOS")
+                       ]))
+        )
+    }
+
+    func test_load_without_authors() throws {
```
Comment (daveverwer) [inline Tests/SPIManifestTests/ManifestTests.swift:73]: I'm not 100% sure what this test is intended to test. The other tests that load manifests without `metadata` or `authors` test that the whole key can be omitted. Is the purpose of this test to ensure that we get `nil` if people add the keys but without data?

If that's the case maybe `test_load_nilAuthors` might be a better name? I'm not sure we need to test this case though.

**Review by daveverwer — CHANGES_REQUESTED** [review]
Just a couple of small comments!

File: Tests/SPIManifestTests/ManifestTests.swift (hunk 3 above)
Comment (jcubit) [reply Tests/SPIManifestTests/ManifestTests.swift:73]: you are right. I think we do not to need this test

**Review by daveverwer — APPROVED** [review]
Thanks so much, @jcubit! This is great 👍

**Pull request merged**